tests/platform_helper/providers/test_yaml_file_provider.py

```diff
@@ -0,0 +1,42 @@
+from pathlib import Path
+
+import pytest
+
+from dbt_platform_helper.providers.yaml_file import DuplicateKeysException
+from dbt_platform_helper.providers.yaml_file import FileNotFoundException
+from dbt_platform_helper.providers.yaml_file import InvalidYamlException
+from dbt_platform_helper.providers.yaml_file import YamlFileProvider
+
+
+def test_returns_valid_yaml(fs):
```

Comment:
probably overkill but could we parameterise these tests with `test_path`, `file_contents`, `valid`, `expected_result` (dictionary or exception)?